components/CreatePost.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+
+type NewPost = {
+  titleDescription: string;
+  type: string;
+};
+
+const iniState = {
+  titleDescription: "",
+  type: "Life",
+};
+
+export const CreatePost = () => {
+  const POST_TYPES = ["Life", "Job", "Joke", "Animals", "Food"];
+  const [isHidden, setIsHidden] = useState(true);
+  const [newPost, setNewPost] = useState<NewPost>(iniState);
+
+  const isEmpty = (str: string) => {
+    const length = str.trim().length;
+    return length < 1;
+  };
+
+  const handleToggleCreatePost = () => {
+    setIsHidden((prev) => !prev);
+  };
+
+  const handleOnSubmit = (e: any) => {
```

Comment:
dont use `any`, using any should be last resort, when we use `any` we are not using the capabilities of typescript